analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/ObjectsShouldNotBeDisposedMoreThanOnceBase.cs

```diff
@@ -18,10 +18,41 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using SonarAnalyzer.SymbolicExecution.Constraints;
+
 namespace SonarAnalyzer.SymbolicExecution.Roslyn.RuleChecks;
 
 public abstract class ObjectsShouldNotBeDisposedMoreThanOnceBase : SymbolicRuleCheck
 {
     protected const string DiagnosticId = "S3966";
-    protected const string MessageFormat = "Refactor this code to make sure '{0}' is disposed only once.";
+    protected const string MessageFormat = "Resource '{1}' {0}";
+    protected const string MessageDisposeTwice = "has already been disposed. Refactor the code to dispose it only once.";
+    protected const string MessageDisposeUsing = "is ensured to be disposed by this using statement. You don't need to dispose it twice.";
```

Comment:
I think it's better to have two messages - as it's not immediately clear that we are detecting the dispose inside the using.

Is there a better way to do it that the one I implemented here?

Reply:
I will revert - update the message  and open an issue.

Reply:
Done - issue opened https://github.com/SonarSource/sonar-dotnet/issues/7357
I added it in the current sprint - we can remove it upon cutting branches.

Reply:
You can just add checkboxes into the original migration issue too. To avoid polluting the backlog with implementation details